app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -287,23 +327,34 @@ object DialogUtils {
 /**
  * Function that will search for if the default hint is correct or needs to be changed.
  */
-private fun createLayerHint(context: Context, layerList: List<String>): String {
+private fun createLayerHint(context: Context, layerList: List<LayerViewModel>): String {
     var hintIndex = 1
     var currentHint = context.getString(R.string.layerHint, hintIndex)
-    while (layerList.contains(currentHint)) {
-        hintIndex += 1
-        currentHint = context.getString(R.string.layerHint, hintIndex)
+    var hintFound = false
+    while (!hintFound) {
+        for (i in layerList.indices) {
+            if (layerList[i].layerName == currentHint) {
+                hintIndex += 1
+                currentHint = context.getString(R.string.layerHint, hintIndex)
+                break
+            } else if (i == layerList.size - 1) {
+                hintFound = true
+                break
+            }
```

Comment:
Boolean doesn't make sense. Can loop on contains instead. You're only looking for the first index of a hint that doesn't exist, so the layerList indices don't matter either.

```
var hintIndex = 0
var currentHint = "" // No need to make this nullable
do {
    hintIndex += 1
    currentHint = context.getString(R.string.layerHint, hintIndex)
} while(layerList.any { it.layerName == currentHint }) // This is the critical simplification
```

Reply:
Going to include this in a cleanup ticket in the future for the app, I knew there was some way to do this, but I couldn't seem to figure it out, since I basically was scanning for a string before, I could just do it again but would need to access it in the layer object.